lesson_14/vlad_butulan/challenge.py

```diff
@@ -0,0 +1,49 @@
+class Matrix(object):
+    def __init__(self, lst):
+        self.lst = lst
+        self.x_dim = len(self.lst[0])
+        self.y_dim = len(self.lst)
+        assert all([self.x_dim == len(line) for line in lst]), "input list is not a matrix"
+    
+    def size(self):
+        return f'M({self.y_dim}, {self.x_dim})'
+
+    def __str__(self):
+        result = ""
+        for line in self.lst:
+            result += ("{}\n".format(line))
+        return result
+
+    def __repr__(self):
+        return str(self.lst)
+    
+    def __add__(self, other):
+        assert self.x_dim == other.x_dim and self.y_dim == other.y_dim, "Incompatible matrixes for +"

```

Comment:
It is not advised to use assert in code. The reason behind this, assert is rising an assert exception. In the case you have here it will be better to rise a different exception than assert exception.

```
if self.x_dim == other.x_dim and self.y_dim == other.y_dim:
    raise ValueError("Incompatible matrixes for +")
```

Reply:
fixed in v1.1